### ISSUE TITLE
[a11y]: Content not within a landmark element

### ISSUE THREAD

**Issue by Mahalakshmivignesh** [carbon-design-system/carbon#16946]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

1.58.1

### React version

1.58.1

### Automated testing tool and ruleset

IBM Equal accessibility checker

### Description

While we have the action options open and run the accessibility tool, i am getting 'content not within landmark' error. It was supposed to be fixed as per the original issue https://github.com/IBMa/equal-access/issues/238. But looks like the issue is still present. The issue https://github.com/IBMa/equal-access/issues/1930 is created under accessibility repo for the same error.

Additional info from the above issue:

> This issue is a little different than the original issue. The original issue was fixed (the owned content is not flagged). However, the focus sentinels are being flagged.
> 
> Add role='main' to the storybook root in this example and then open the menu and scan to see the issue:
> https://react.carbondesignsystem.com/iframe.html?args=&id=components-overflowmenu--default&viewMode=story
> 
> The actual menu works fine since it is referenced by the opening button. However, the focus sentinels are flagged.
> 
> Unfortunately, I'm not sure if we have a robust way of identifying those as focus sentinels (other than a Carbon-specific implementation).

> ![image](https://github.com/carbon-design-system/carbon/assets/141624666/cef840e8-9e3d-4db1-9978-e2b4d40a1873)

### Reproduction/example

Refer below steps to reproduce 

### Steps to reproduce

test case: https://react.carbondesignsystem.com/iframe.html?args=&id=components-overflowmenu--default&viewMode=story. Add a role='main' to the storybook root, then click the three-dot button, scan again and the issue appears.

### Suggested Severity

None

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
@Mahalakshmivignesh If the `<body>` does not have a landmark, you'll get this violation because 

> `OverflowMenu` uses React Portals to render the overflow menu into the DOM. To determine where in the DOM the menu will be placed, it looks for a parent element with the `data-floating-menu-container` attribute. If no parent with this attribute is found, the menu will be placed on `document.body`.

This is from the [documentation](https://react.carbondesignsystem.com/?path=/docs/components-overflowmenu--overview#data-floating-menu-container) in storybook. To fix you can either add a landmark role to the `<body>` or use `data-floating-menu-container` to ensure it's rendered into portion of the DOM already wrapped by a landmark role. [Here's an example](https://stackblitz.com/edit/github-u7nfga?file=src%2FApp.jsx).

Additionally, if you're running into issues specifically with the focus sentinels, we have a new focus wrap implementation that does not use focus sentinels. It's behind the `enable-experimental-focus-wrap-without-sentinels` flag. We believe it to be a stable robust implementation, but still placed it behind a flag out of caution because we'd like to get feedback ensuring it works as you'd expect. If you turn it on, please let us know your feedback.

**Comment by tay1orjones** [comment]
Thinking on this a bit more, It probably makes the most sense in nearly all situations to use `data-floating-menu-container` to add it to an existing landmark on the page, such as a `main` tag. We unfortunately can't do this by default in the component because there is no guarantee a `main` tag is on the page it's being used in. So it defaults to `body`

Docs on the feature flags are in [storybook](https://react.carbondesignsystem.com/?path=/docs/experimental-feature-flags--overview)
